astrokat/utility.py

```diff
@@ -27,6 +27,11 @@ def read_yaml(filename):
         # not a yaml file, suspected csv file, returning False
         return {}
 
+    # remove empty keys
+    for key in data.keys():
+        if data[key] is None:
+            del data[key]
```

Comment:
This works in Python 2, but not in Python 3.  In general, it is a bad idea to change the container that is being iterated over.   Changing the loop to use `for key in list(data.keys()):` should work better.

Example of failure in Python 3
```
$ ipython
Python 3.6.3 (default, Oct  4 2017, 06:09:05) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.2.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: d=dict(a=1,b=None)

In [2]: for key in d.keys():
   ...:     if d[key] is None:
   ...:         del d[key]
   ...:         
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-2-fdab914f0c57> in <module>()
----> 1 for key in d.keys():
      2     if d[key] is None:
      3         del d[key]
      4 

RuntimeError: dictionary changed size during iteration

In [3]: 
```

Reply:
Sorry, bad implementation, corrected